pulsar-broker/src/test/java/org/apache/pulsar/broker/namespace/NamespaceUnloadingTest.java

```diff
@@ -68,4 +75,26 @@ public void testUnloadPartiallyLoadedNamespace() throws Exception {
         producer.close();
     }
 
+    @Test
+    public void testUnloadWithTopicCreation() throws PulsarAdminException, PulsarClientException {
+        final String namespaceName = "prop/ns_unloading";
+        final String topicName = "persistent://prop/ns_unloading/with_topic_creation";
+        final int partitions = 5;
+        admin.namespaces().createNamespace(namespaceName, 1);
+        admin.topics().createPartitionedTopic(topicName, partitions);
+        @Cleanup
+        Producer<Integer> producer = pulsarClient.newProducer(Schema.INT32)
+                .topic(topicName)
+                .create();
+
+        for (int i = 0; i < 100; i++) {
+            admin.namespaces().unloadNamespaceBundle(namespaceName, "0x00000000_0xffffffff");
+        }
+
+        for (int i = 0; i < partitions; i++) {
+            producer.send(i);
+        }
+        admin.namespaces().deleteNamespace(namespaceName, true);
```

Comment:
The producer should be closed before deleting the namespace.

Reply:
I've got two reasons to write like this. 

- One advantage of using @Cleanup is that it ensures resources are released no matter what. For instance, if I call the close method on the producer first and it throws an exception, subsequent operations won't be executed, potentially leading to resource leaks in a testing environment.

- In order to achieve our goals, it's advisable to avoid writing tests in a routine manner, as we are dealing with a public API. As long as it can be invoked successfully, users will adopt this approach, and we should ideally cover any subsequent consequences or impacts as well.